### PR TITLE
use minfee if purchase slots are not full

### PR DESCRIPTION
**Pull request by raedah**
closes https://github.com/decred/dcrticketbuyer/issues/23

### CODE REVIEW

File: ticketbuyer/buyer.go [hunk 1]
```diff
@@ -378,46 +353,57 @@ func (t *TicketPurchaser) Purchase(height int64) (*PurchaseStats, error) {
...
-	if feeToUse > t.cfg.MaxFee {
-		log.Infof("Not buying because max fee exceed: (max fee: %.8f DCR,  scaled fee: %.8f DCR)",
-			t.cfg.MaxFee, feeToUse)
-		return ps, nil
-	}
-	if feeToUse < t.cfg.MinFee {
-		log.Debugf("Using min ticket fee: %.8f DCR (scaled fee: %.8f DCR)", t.cfg.MinFee, feeToUse)
+	var feeToUse float64
+	if ticketPurchasesInLastBlock < uint32(t.activeNet.MaxFreshStakePerBlock) {
+		log.Debugf("Using min ticket fee: %.8f DCR (last block purchase slots not filled)", t.cfg.MinFee)
```
Comment (raedah) [inline ticketbuyer/buyer.go]: These few lines above are really all I added. The rest of the code blocks are reused and just moved in position.

File: ticketbuyer/buyer.go [hunk 2]
```diff
@@ -161,11 +160,6 @@ type PurchaseStats struct {
 	MempoolOwn    int
 	Purchased     int
 	LeftWindow    int
-	FeeMin        float64
-	FeeMax        float64
-	FeeMedian     float64
-	FeeMean       float64
-	FeeOwn        float64
```
Comment (raedah) [inline ticketbuyer/buyer.go]: these are not used, removing

File: ticketbuyer/buyer.go [hunk 3]
```diff
@@ -378,46 +351,73 @@ func (t *TicketPurchaser) Purchase(height int64) (*PurchaseStats, error) {
...
-		if err != nil {
-			return ps, err
-		}
+	// Lookup how many tickets purchase slots were filled in the last block
+	oneBlock := uint32(1)
+	info, err := t.dcrdChainSvr.TicketFeeInfo(&oneBlock, &zeroUint32)
+	if err != nil {
+		return ps, err
 	}
+	ticketPurchasesInLastBlock := int(info.FeeInfoBlocks[0].Number)
```
Comment (alexlyp) [inline ticketbuyer/buyer.go]: need to check info.FeeInfoBlocks len before referring to [0]

File: version.go [hunk 4]
```diff
@@ -27,9 +27,9 @@ const (
 )
 
 // appBuild is defined as a variable so it can be overridden during the build
-// process with '-ldflags "-X main.appBuild foo' if needed.  It MUST only
+// process with '-ldflags "-X main.appBuild=foo' if needed.  It MUST only
 // contain characters from semanticAlphabet per the semantic versioning spec.
-var appBuild string
+var appBuild = "dev"
```
Comment (alexlyp) [inline version.go]: Remove this, submit a different PR if you want to change this

Comment (jcvernaleo) [inline version.go]: Must be some rebasing needed because that change was one of mine from #578 and is already in master.

Comment (raedah) [inline version.go]: yeah, was rebase

File: ticketbuyer/buyer.go (hunk 3 above)
Comment (raedah) [inline ticketbuyer/buyer.go]: thats in

File: ticketbuyer/buyer.go [hunk 5]
```diff
@@ -378,46 +351,76 @@ func (t *TicketPurchaser) Purchase(height int64) (*PurchaseStats, error) {
...
-		if err != nil {
-			return ps, err
-		}
+	// Lookup how many tickets purchase slots were filled in the last block
+	oneBlock := uint32(1)
+	info, err := t.dcrdChainSvr.TicketFeeInfo(&oneBlock, &zeroUint32)
+	if err != nil {
+		return ps, err
+	}
+	if len(info.FeeInfoBlocks) != 1 {
```
Comment (alexlyp) [inline ticketbuyer/buyer.go]: > than 0 is probably better here
